test/test_matplotlib.py

```diff
@@ -169,7 +176,189 @@ def do_plot(ax, Z, transform):
+        plt.axes([0.01, 0.01, 0.98, 0.90], facecolor="white", frameon=True)
+        plt.gca().set_xlim(0., 1.)
+        plt.gca().set_ylim(0., 1.)
+        plt.gca().set_title("Matplotlib's math rendering engine",
+                            color=mpl_grey_rvb, fontsize=14, weight='bold')
+        plt.gca().set_xticklabels("", visible=False)
+        plt.gca().set_yticklabels("", visible=False)
+
+        # Gap between lines in axes coords
+        line_axesfrac = (1. / (n_lines))
+
+        # Plotting header demonstration formula
+        full_demo = mathext_demos[0]
+        plt.annotate(full_demo,
+                    xy=(0.5, 1. - 0.59 * line_axesfrac),
+                    color=mpl_orange_rvb, ha='center', fontsize=20)
+
+        # Plotting features demonstration formulae
+        for i_line in range(1, n_lines):
+            baseline = 1 - (i_line) * line_axesfrac
+            baseline_next = baseline - line_axesfrac
+            title = mathtext_titles[i_line] + ":"
+            fill_color = ['white', mpl_blue_rvb][i_line % 2]
+            plt.fill_between([0., 1.], [baseline, baseline],
+                            [baseline_next, baseline_next],
+                            color=fill_color, alpha=0.5)
+            plt.annotate(title,
+                        xy=(0.07, baseline - 0.3 * line_axesfrac),
+                        color=mpl_grey_rvb, weight='bold')
+            demo = mathext_demos[i_line]
+            plt.annotate(demo,
+                        xy=(0.05, baseline - 0.75 * line_axesfrac),
+                        color=mpl_grey_rvb, fontsize=16)
+
+        for i in range(n_lines):
+            s = mathext_demos[i]
+            print(i, s)
+        plt.show()
+    doall()
+    """)
+
+    check_comparison(selenium, 'canvas-math-text')
+
+
+def test_custom_font_text(selenium_standalone):
+    selenium = selenium_standalone
+    selenium.load_package("matplotlib")
+    selenium.run("""
+    from js import window
+    window.testing = True
+    import matplotlib.pyplot as plt
+    import numpy as np
+
+    f = {'fontname': 'cmsy10'}
+
+    t = np.arange(0.0, 2.0, 0.01)
+    s = 1 + np.sin(2 * np.pi * t)
+    plt.figure()
+    plt.title('A simple Sine Curve', **f)
+    plt.plot(t, s, linewidth=1.0, marker=11)
+    plt.plot(t, t)
+    plt.grid(True)
+    plt.show()
+    """)
+
+    check_comparison(selenium, 'canvas-custom-font-text', load_font=True)
+
+
 class ResultLoaded:
     def __call__(self, driver):
         inited = driver.execute_script("return window.result")
         return inited is not None
+
+
+class FontsLoadedFirefox:
+    def __call__(self, driver):
+        font_inited = driver.execute_script("return window.font_counter")
+        return font_inited is not None and font_inited == 38
+
+
+class FontsLoadedChrome:
+    def __call__(self, driver):
+        font_inited = driver.execute_script("return window.font_counter")
+        return font_inited is not None and font_inited == 56
```

Comment:
Interesting.  A value of 56 doesn't really make much sense to me, since there are only 42 fonts in that matplotlib set that we're loading from.  Can you investigate what's in the `_font_set` variable to see what's being loaded?